rules/integrations/azure/credential_access_entra_id_brute_force_activity.toml

```diff
@@ -0,0 +1,244 @@
+
+- Legitimate automation (e.g., misconfigured scripts, sync processes) can trigger repeated failures.
+- Internal red team activity or penetration tests may mimic brute-force behaviors.
+- Certain service accounts or mobile clients may generate repetitive sign-in noise if not properly configured.
+
+### Response and remediation
+
+- Notify your identity security team for further analysis.
+- Investigate and lock or reset impacted accounts if compromise is suspected.
+- Block offending IPs or ASNs at the firewall, proxy, or using Conditional Access.
+- Confirm MFA and Conditional Access are enforced for all user types.
+- Audit targeted accounts for credential reuse across services.
+- Implement account lockout or throttling for failed sign-in attempts where possible.
+"""
+references = [
+    "https://www.microsoft.com/en-us/security/blog/2025/05/27/new-russia-affiliated-actor-void-blizzard-targets-critical-sectors-for-espionage/",
+    "https://cloud.hacktricks.xyz/pentesting-cloud/azure-security/az-unauthenticated-enum-and-initial-entry/az-password-spraying",
+    "https://learn.microsoft.com/en-us/security/operations/incident-response-playbook-password-spray",
+    "https://learn.microsoft.com/en-us/purview/audit-log-detailed-properties",
+    "https://securityscorecard.com/research/massive-botnet-targets-m365-with-stealthy-password-spraying-attacks/",
+    "https://learn.microsoft.com/en-us/entra/identity-platform/reference-error-codes",
+    "https://github.com/0xZDH/Omnispray",
+    "https://github.com/0xZDH/o365spray",
+]
+risk_score = 47
+rule_id = "cca64114-fb8b-11ef-86e2-f661ea17fbce"
+severity = "medium"
+tags = [
+    "Domain: Cloud",
+    "Domain: Identity",
+    "Data Source: Azure",
+    "Data Source: Entra ID",
+    "Data Source: Entra ID Sign-in Logs",
+    "Use Case: Identity and Access Audit",
+    "Use Case: Threat Detection",
+    "Tactic: Credential Access",
+    "Resources: Investigation Guide",
+]
+timestamp_override = "event.ingested"
+type = "esql"
+
+query = '''
+FROM logs-azure.signinlogs*
+
+| EVAL
+    time_window = DATE_TRUNC(15 minutes, @timestamp),
+    user_id = TO_LOWER(azure.signinlogs.properties.user_principal_name),
+    ip = source.ip,
+    login_error = azure.signinlogs.result_description,
+    error_code = azure.signinlogs.properties.status.error_code,
+    request_type = TO_LOWER(azure.signinlogs.properties.incoming_token_type),
+    app_name = TO_LOWER(azure.signinlogs.properties.app_display_name),
+    asn_org = source.`as`.organization.name,
+    country = source.geo.country_name,
+    user_agent = user_agent.original,
+    event_time = @timestamp
+
+| WHERE event.dataset == "azure.signinlogs"
+    AND event.category == "authentication"
+    AND azure.signinlogs.category IN ("NonInteractiveUserSignInLogs", "SignInLogs")
+    AND event.outcome == "failure"
+    AND azure.signinlogs.properties.authentication_requirement == "singleFactorAuthentication"
+    AND error_code != 50053
```

Comment:
I'm not sure this exclusion is necessary right? since you include the error_codes you want to evaluate in the next section

Reply:
Good catch. Removed!